test/index.spec.js

```diff
@@ -236,3 +240,63 @@ it('should fulfill the promise the array of expected actions is contained in the
 
     return promise;
 });
+
+it('should teardown correctly when promise fulfills', async () => {
+    const mockStore = createMockStore([thunkMiddleware]);
+    const clearTimeoutSpy = jest.spyOn(global, 'clearTimeout');
+    const getUnsubcribeSpy = spyOnUnsubscribe(mockStore);
+
+    mockStore.dispatch(asyncActionsCreator([action1]));
+
+    await waitForActions(mockStore, action1);
+
+    expect(clearTimeoutSpy).toHaveBeenCalledTimes(1);
+    expect(getUnsubcribeSpy()).toHaveBeenCalledTimes(1);
+});
+
+it('should teardown correctly when promise rejects via timeout', async () => {
+    expect.assertions(1);
+    const mockStore = createMockStore([thunkMiddleware]);
+    const getUnsubcribeSpy = spyOnUnsubscribe(mockStore);
+
+    mockStore.dispatch(asyncActionsCreator([action1]));
+
+    try {
+        await waitForActions(mockStore, [action1, action2]);
+    } catch (err) {
+        expect(getUnsubcribeSpy()).toHaveBeenCalledTimes(1);
+    }
+});
+
+it('should teardown correctly when promise rejects via cancelation', () => {
+    expect.assertions(2);
+    const mockStore = createMockStore([thunkMiddleware]);
+    const clearTimeoutSpy = jest.spyOn(global, 'clearTimeout');
+    const getUnsubcribeSpy = spyOnUnsubscribe(mockStore);
+
+    mockStore.dispatch(asyncActionsCreator([action1]));
+
+    const promise = waitForActions(mockStore, [action1, action2]);
+
+    promise.catch(() => {
```

Comment:
I think this test is wrong because it's async but we are not awaiting nor returning the promise.

Could use await?

```js
setTimeout(() => promise.cancel(), 10);

try {
    await promise;
} catch (err) {
    // expectations here
}
```

Reply:
Yeah, I agree, it looks weird. I'll fix it.